projects/pgai/pgai/cli.py

```diff
@@ -770,30 +770,119 @@ def load(
             handlers=log_handlers,
         )
 
+    from rich.console import Console
+
+    console = Console(stderr=True, quiet=quiet)
+
     catalog_name = catalog_name or "default"
-    batch_size = batch_size if batch_size is not None else 32
-    sql_file = sql_file.expanduser().resolve()
-    assert sql_file and sql_file.exists() and sql_file.is_file(), "invalid sql file"
+    yaml_file = yaml_file.expanduser().resolve()
+    assert yaml_file.is_file(), "invalid yaml file"
+
+    async def do():
+        from pgai.semantic_catalog import from_name
+
+        async with (
+            await psycopg.AsyncConnection.connect(catalog_db_url) as ccon,
+            await psycopg.AsyncConnection.connect(db_url) as tcon,
+        ):
+            console.status(f"finding '{catalog_name}' catalog...")
+            sc = await from_name(ccon, catalog_name)
+            with yaml_file.open(mode="r") as r:
+                await sc.import_catalog(
+                    ccon, tcon, r, embed_config, batch_size, console
+                )
+
+    asyncio.run(do())
 
-    script = sql_file.read_text()
+
+@semantic_catalog.command(name="export")
+@click.option(
+    "-c",
+    "--catalog-db-url",
+    type=click.STRING,
+    default="postgres://postgres@localhost:5432/postgres",
+    show_default=True,
+    help="The connection URL to the database the semantic catalog is in.",
+    envvar="CATALOG_DB",
+)
+@click.option(
+    "-f",
+    "--yaml-file",
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+    default=None,
+    help="The path to a yaml file.",
+)
+@click.option(
+    "-n",
+    "--catalog-name",
+    type=click.STRING,
+    default="default",
+    help="The name of the semantic catalog to generate embeddings for.",  # noqa: E501
+)
+@click.option(
+    "-q",
+    "--quiet",
+    is_flag=True,
+    help="Do not print log messages.",
+)
+@click.option(
+    "-l",
+    "--log-file",
+    type=click.Path(dir_okay=False, writable=True, path_type=Path),
+    default=None,
+    help="The path to a file to write log messages to.",
+)
+@click.option(
+    "--log-level",
+    type=click.Choice(
+        ["DEBUG", "INFO", "WARN", "ERROR", "FATAL", "CRITICAL"], case_sensitive=False
+    ),
+    default="INFO",
+)
+def export_catalog(
```

Comment:
Maybe add an option to export only certain objects from the catalog?

Reply:
yeah, let's do that in a follow-up. I'll add a ticket